src/lib/drivers/device/nuttx/SPI.cpp

```diff
@@ -90,90 +79,56 @@ SPI::~SPI()
 int
 SPI::init()
 {
-	int ret = OK;
-
 	/* attach to the spi bus */
 	if (_dev == nullptr) {
 		int bus = get_device_bus();
 
 		if (!board_has_bus(BOARD_SPI_BUS, bus)) {
-			ret = -ENOENT;
-			goto out;
+			return -ENOENT;
 		}
 
 		_dev = px4_spibus_initialize(bus);
 	}
 
 	if (_dev == nullptr) {
 		DEVICE_DEBUG("failed to init SPI");
-		ret = -ENOENT;
-		goto out;
+		return -ENOENT;
 	}
 
 	/* deselect device to ensure high to low transition of pin select */
 	SPI_SELECT(_dev, _device, false);
 
 	/* call the probe function to check whether the device is present */
-	ret = probe();
+	int ret = probe();
 
 	if (ret != OK) {
 		DEVICE_DEBUG("probe failed");
-		goto out;
+		return ret;
 	}
 
 	/* do base class init, which will create the device node, etc. */
 	ret = CDev::init();
 
 	if (ret != OK) {
 		DEVICE_DEBUG("cdev init failed");
-		goto out;
+		return ret;
 	}
 
 	/* tell the workd where we are */
 	DEVICE_LOG("on SPI bus %d at %d (%u KHz)", get_device_bus(), PX4_SPI_DEV_ID(_device), _frequency / 1000);
 
-out:
-	return ret;
+	return PX4_OK;
 }
 
 int
 SPI::transfer(uint8_t *send, uint8_t *recv, unsigned len)
 {
-	int result;
-
 	if ((send == nullptr) && (recv == nullptr)) {
 		return -EINVAL;
 	}
 
-	LockMode mode = up_interrupt_context() ? LOCK_NONE : locking_mode;
-
-	/* lock the bus as required */
-	switch (mode) {
-	default:
-	case LOCK_PREEMPTION: {
-			irqstate_t state = px4_enter_critical_section();
-			result = _transfer(send, recv, len);
-			px4_leave_critical_section(state);
-		}
-		break;
-
-	case LOCK_THREADS:
-		SPI_LOCK(_dev, true);
-		result = _transfer(send, recv, len);
-		SPI_LOCK(_dev, false);
-		break;
-
-	case LOCK_NONE:
-		result = _transfer(send, recv, len);
-		break;
-	}
-
-	return result;
-}
+	SPI_LOCK(_dev, true);
```

Comment:
Please  add an assertion on  !up_interrupt_context() so this is not misinterpreted.